### PR TITLE
Allow Travis CI to build on PHP 7.4

### PR DESCRIPTION
**Pull request by phansys**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by Tobion** [comment]
This needs to target 3.4

**Comment by phansys** [comment]
> This needs to target 3.4

Target updated :+1:

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
I'm putting this on hold because we usually don't add "allowed_job" lines: this would consume a job for every PR+merge, while they are quite precious.
Of course, once it's green and reasonably stable, we'll merge as a regular job.
Then, we'll want to rotate all jobs: deps=low should be the one running 7.4, deps=high 7.3, etc.
Could you please have a look to make this work?

**Comment by phansys** [comment]
Of course!
I'll try to check how to fix the job, since the issue seems to be APC related ([ref.](https://travis-ci.community/t/add-php-7-4-branch-to-test-against/2179/4)).

**Comment by phansys** [comment]
The tests are running now. The failures are caused by unsilenced deprecation notices.

**Comment by nicolas-grekas** [comment]
> Function ReflectionType::__toString() is deprecated

this happened to Twig today too and [was fixed by removing the use of some mocks](https://github.com/twigphp/Twig/pull/3081)
Not sure we can do it for Symfony.
Instead, we could try making the `phpunit` script (see repo root) use phpunit 8.2 on PHP 7.4.

**Comment by nicolas-grekas** [comment]
We should mute this specific deprecation in the bridge I think, because many projects will be affected. /cc @greg0ire

File: phpunit
Comment (fabpot) [inline phpunit]: What about 8.2 here?

**Comment by greg0ire** [comment]
> We should mute this specific deprecation in the bridge I think, because many projects will be affected. /cc @greg0ire

:thinking: 	maybe we could add a new deprecation group called `muted` and have a way to display them anyway, just in case? `SYMFONY_DEPRECATIONS_HELPER="scream=true"`?

File: phpunit
Comment (phansys) [reply phpunit]: Why not... https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-8.2.md#820---2019-06-07

**Comment by phansys** [comment]
PHPUnit ^8.0 seems to require some missing return declarations at inherited methods, by instance:
```
Declaration of Symfony\Bundle\WebProfilerBundle\Tests\DependencyInjection\WebProfilerExtensionTest::setUp() must be compatible with PHPUnit\Framework\TestCase::setUp(): void
```
Given that, I think this version bump can't be done in a minor release. Please, let me know how to proceed if you have elaborated some concerns about this.

**Comment by greg0ire** [comment]
> Given that, I think this version bump can't be done in a minor release. 

You might be able to, but it will take some black magic:

https://github.com/symfony/symfony/blob/6811aaa8e087665769dd66c69fb3109d002196ef/src/Symfony/Bridge/PhpUnit/CoverageListener.php#L14-L20

**Comment by greg0ire** [comment]
[I built a thing!](https://github.com/greg0ire/symfony/commit/ca675239922a4f335091145c09d66b4e9ec6afd6) I'm a bit unsure if this should be considered bugfix or feature, and what version of the phpunit bridge is used to test 3.4 . Please advise.

**Comment by nicolas-grekas** [comment]
@greg0ire sorry I should have been more specific: we should mute based on the message AND the stack trace: only deprecations from phpunit should be muted.

**Comment by greg0ire** [comment]
Oh ok it makes a lot more sense now ^^

**Comment by greg0ire** [comment]
I ended up picking 4.3, see link above.

**Comment by phansys** [comment]
@greg0ire, here is a sample build on `4.3` after the merge of #32443: https://travis-ci.org/phansys/symfony/jobs/560515287.

**Comment by nicolas-grekas** [comment]
Needs https://github.com/egulias/EmailValidator/pull/203 now

**Comment by greg0ire** [comment]
> @greg0ire, here is a sample build on 4.3 after the merge of #32443: https://travis-ci.org/phansys/symfony/jobs/560515287.

Oh wow my fix works already? I thought it would be necessary to have it land in the `vendor` directory, which implies a new release… :thinking: Anyway, cool :)

**Comment by nicolas-grekas** [comment]
Next step unlocked. We could replace all setup/teardown method by before/after ones, with the same as annotations.

**Review by nicolas-grekas — APPROVED** [review]
I'm now merging this on 3.4 only so that we can have some visibility on PHP7.4 support without consuming too many jobs.

**Comment by nicolas-grekas** [comment]
Thank you @phansys.

**Pull request merged**
